hack/tools/release/notes.go

```diff
@@ -509,3 +509,42 @@ func generateReleaseNoteEntry(c *commit) (*releaseNoteEntry, error) {
 func trimReleaseBackportMarker(title string) string {
 	return releaseBackportMarker.ReplaceAllString(title, "${1}")
 }
+
+// lessStringCaseInsensitive implements the Less signature for strings with case insensitivity.
+// It can be used with the standard library sort utils.
```

Comment:
Let's please add a comment that we just did this for fun / performance optimization.

To be honest for maintainability I would have preferred the previous version if we don't have a performance issue with sorting

Reply:
Ah I see. you are correct, this is only for performance and there is no perceivable difference for a human, the time spent calling Github's API is orders of magnitude higher. If you feel strongly, I can leave it as it was.

Reply:
Agreed - let's use `sort.Slice` here to help future release teams understand what's going on with this sorting.

Reply:
done